core/src/main/java/org/springframework/security/provisioning/InMemoryUserDetailsManager.java

```diff
@@ -143,7 +143,7 @@ public void changePassword(String oldPassword, String newPassword) {
 	@Override
 	public UserDetails updatePassword(UserDetails user, String newPassword) {
 		String username = user.getUsername();
-		MutableUserDetails mutableUser = this.users.get(username);
+		MutableUserDetails mutableUser = this.users.get(username.toLowerCase());
```

Comment:
If `username` is itself `null`, `username.toLowerCase()` will cause `NullPointerException`

Reply:
`username` can't be `null`.

Reply:
Considering overall theme of this class, `null` check is not required. I guess when `null` is passed, `NullPointerException` is expected outcome.